commandapi-platforms/commandapi-bukkit/commandapi-bukkit-test/commandapi-bukkit-test-tests/src/test/java/dev/jorel/commandapi/test/ConfigGenerationTests.java

```diff
@@ -0,0 +1,203 @@
+			"If \"true\", outputs command registration and unregistration logs in the console"
+		}, false);
+		missingExecutorImplementation = new CommentedConfigOption<>(new String[]{
+			"Missing executor implementation (default: \"This command has no implementations for %s\")",
+			"The message to display to senders when a command has no executor. Available",
+			"parameters are:",
+			"  %s - the executor class (lowercase)",
+			"  %S - the executor class (normal case)"
+		}, "This command has no implementations for %s");
+
+		messages = new CommentedSection(new String[]{
+			"Messages",
+			"Controls messages that the CommandAPI displays to players"
+		});
+
+		Map<String, CommentedConfigOption<?>> options = new LinkedHashMap<>();
+		options.put("silent-logs", silentLogs);
+		options.put("verbose-outputs", verboseOutputs);
+		options.put("messages.missing-executor-implementation", missingExecutorImplementation);
+
+		Map<String, CommentedSection> sections = new LinkedHashMap<>();
+		sections.put("messages", messages);
+
+		ConfigurationAdapter<YamlConfiguration, DefaultBukkitConfig> adapter = new BukkitConfigurationAdapter(new YamlConfiguration());
+		bukkitConfig = DefaultBukkitConfig.create(options, sections);
+		generator = ConfigGenerator.createNew(bukkitConfig);
+		this.adapter = (BukkitConfigurationAdapter) generator.generate(adapter);
+	}
+
+	@AfterEach
+	public void reset() {
+		this.silentLogs = null;
+		this.verboseOutputs = null;
+		this.missingExecutorImplementation = null;
+		this.messages = null;
+		this.generator = null;
+		this.bukkitConfig = null;
+		this.adapter = null;
+	}
+
+	// Test methods
+	private void validateConfigOptions(Set<String> options, BukkitConfigurationAdapter adapter) {
+		for (String option : options) {
+			assertTrue(adapter.contains(option), "Config option '" + option + "' does not exist!");
+		}
+	}
+
+	private void validateConfigOptionComments(Map<String, String[]> comments, BukkitConfigurationAdapter adapter) {
+		for (String option : comments.keySet()) {
+			String[] expectedComment = comments.get(option);
+			String[] generatedComment = adapter.getComment(option);
+			assertArrayEquals(expectedComment, generatedComment, "Config option comment for option '" + option + "' does not exist or was incorrect!");
+		}
+	}
+
+	private void validateConfigOptionsAbsent(Set<String> options, BukkitConfigurationAdapter adapter) {
+		for (String option : options) {
+			assertFalse(adapter.contains(option), "Config option '" + option + "' does still exist!");
+		}
+	}
+
+	private void validateSections(List<String> sections, String expectedOption, YamlConfiguration config) {
+		ConfigurationSection root = config.getConfigurationSection(sections.get(0));
+		assertNotNull(root, "Section '" + sections.get(0) + "' does not exist!");
+
+		for (int i = 1; i < sections.size(); i++) {
+			root = root.getConfigurationSection(sections.get(i));
+			assertNotNull(root, "Section '" + sections.get(i) + "' does not exist!");
+		}
```

Comment:
The case for the first element could be included in the loop, but it works the same :shrug:.

```suggestion
		ConfigurationSection root = config;
		for (String section : sections) {
			root = root.getConfigurationSection(section);
			assertNotNull(root, "Section '" + section + "' does not exist!");
		}
```